cloudformation/README.md

```diff
@@ -0,0 +1,131 @@
+# Cirrus CloudFormation Infrastructure
+
+This directory contains CloudFormation templates for deploying **cirrus-geo**
+infrastructure on AWS.
+
+## Overview
+
+The CloudFormation implementation consists of:
+
+- **Base Infrastructure**: S3 buckets, DynamoDB state table, SQS queues, SNS topics
+- **VPC**: Virtual Private Cloud with public/private subnets across 2 availability zones
+- **Lambda Functions**: 5 core functions (API, Process, Update State, Pre-Batch,
+  Post-Batch) with IAM roles
+- **API Gateway**: REST API (EDGE) with Lambda integration and CloudWatch logging
+- **CLI Configuration**: Parameter Store setup for cirrus CLI deployment discovery
+
+The CloudFormation templates are modular and use nested stacks for organization.
+
+## Directory Structure
+
+```bash
+cloudformation/
+├── main.yaml                # Main template (entry point)
+├── parameters.json          # Parameter file with default values
+├── bootstrap/
+│   └── bootstrap.yaml       # Bootstrap template (S3 bucket for deployment artifacts)
+├── core/                    # Core cirrus-geo infrastructure stacks
+│   ├── base.yaml            # Base infrastructure (S3, DynamoDB, SQS, SNS)
+│   ├── functions.yaml       # Lambda functions with IAM roles
+│   ├── api.yaml             # API Gateway and related resources
+│   ├── vpc.yaml             # VPC and networking (2-AZ setup)
+│   └── lambda-packages/     # Zipped Python Lambda function code
+└── cli/
+    └── ssm_parameters.yaml  # SSM parameters for CLI
+```
+
+## Quick Start
+
+### Prerequisites
+
+1. **AWS CLI** installed and configured
+2. **Python 3.12+** for Lambda function packaging
+3. AWS credentials with appropriate permissions
+
+### Basic Deployment
+
+1. **Deploy bootstrap stack** (creates S3 bucket for deployment artifacts):
+
+   ```bash
+   aws cloudformation create-stack \
+     --stack-name cirrus-bootstrap \
+     --template-body file://cloudformation/bootstrap/bootstrap.yaml \
+     --parameters file://cloudformation/parameters.json
```

Comment:
Looks like only `ResourcePrefix` should be specified here...and the parameter file contains much else:

```shell
aws cloudformation create-stack \
     --stack-name irc-cirrus-bootstrap \
     --template-body file://cloudformation/bootstrap/bootstrap.yaml \
     --parameters file://cloudformation/parameters.json

An error occurred (ValidationError) when calling the CreateStack operation: Parameters: [PrivateSubnet1Cidr, PublicSubnet2Cidr, VpcCidr, PrivateSubnet2Cidr, PublicSubnet1Cidr, LogLevel] do not exist in the template
```

Reply:
Perhaps, this is the best replacement?
```suggestion
   aws cloudformation create-stack \
     --stack-name cirrus-bootstrap \
     --template-body file://cloudformation/bootstrap/bootstrap.yaml \
     --parameters ParameterKey=ResourcePrefix,ParameterValue=cirrus-sandbox
```

Reply:
>One more here, should the cirrus-bootstrap and cirrus-sandbox stack names be environment variables in these instructions? 

Yeah, we should probably make it clear in the instructions that those are "variable" values. Whether that is via an environment variable, or just an indication that the user needs to insert appropriate values in those locations.